iroha-cli/interactive/impl/interactive_common_cli.cpp

```diff
@@ -95,21 +101,28 @@ namespace iroha_cli {
 
     nonstd::optional<std::vector<std::string>> parseParams(
         std::string line, std::string command_name, ParamsMap params_map) {
-      auto params_description = findInHandlerMap(command_name, params_map);
+      auto params_description =
+          findInHandlerMap(command_name, std::move(params_map));
       if (not params_description.has_value()) {
         // Report no params where found for this command
         std::cout << "Command params not found" << std::endl;
         // Stop parsing, something is not implemented
         return nonstd::nullopt;
       }
-      auto words = parser::split(line);
+      auto words = parser::split(std::move(line));
       if (words.size() == 1) {
         // Start interactive mode
         std::vector<std::string> params;
-        std::for_each(
-            params_description.value().begin(),
-            params_description.value().end(),
-            [&params](auto param) { params.push_back(promtString(param)); });
+        std::for_each(params_description.value().begin(),
+                      params_description.value().end(),
+                      [&params](auto param) {
+                        auto val = promtString(param);
+                        if (not val.has_value()) {
+                          // Exit symbol, stop parsing
+                          return nonstd::nullopt;
```

Comment:
That won't work (if I properly understood what did you what), use explicit for-each&break if you want to stop cycling at some point:
```cpp
for (auto param: params_description.value()) {
  auto val = promtString(param);
  if (not val.has_value()) {
    break;
  }
  params.push_back(val.value());
}
```